### ISSUE TITLE
False positive duplicate field when using cfg attribute in struct initializer

### ISSUE THREAD

**Issue by Mcat12** [intellij-rust/intellij-rust#4760]
## Environment

* **IntelliJ Rust plugin version:** 0.2.112.2148-193
* **Rust toolchain version:** 1.39.0
* **IDE name and version:** CLion 2019.3.1
* **Operating system:** Linux (Fedora 31)

## Problem description
When conditionally defining a value in a struct initializer, a duplicate field error is thrown. However, rustc handles the code just fine.

Note that this error only appears for cfg directives that the plugin does not fully recognize (features and test). It does work for `debug_assertions`, `target_family`, and the like.

## Steps to reproduce

```rust
#[derive(Debug)]
struct Foo {
    bar: String
}

fn main() {
    let foo = Foo {
        #[cfg(test)] bar: "testing".to_owned(),
        #[cfg(not(test))] bar: "not testing".to_owned(),
    };

    println!("{:?}", foo);
}
```

![image](https://user-images.githubusercontent.com/4417660/70942603-05b35280-201d-11ea-8c18-5eee10aa31b3.png)

![image](https://user-images.githubusercontent.com/4417660/70942659-24194e00-201d-11ea-98c1-c51f99b79cbf.png)

**Comment by Undin** [comment]
@Mcat12 Thanks for the report!
It's known issue.
> Note that this error only appears for cfg directives that the plugin does not fully recognize (features and test). It does work for debug_assertions, target_family, and the like.

Cargo features don't work only when they are declared in your workspace, features from libs are already supported. But we are working on it right now.
`test` option is tougher because you usually interact with production and test code at the same time and in the ideal case the plugin has to support two mutually exclusive conditions: `test` and `not(test)`. Unfortunately, we don't have a good solution for it what's why we haven't touched `test` option yet

**Comment by Mcat12** [comment]
I originally found the issue when using `feature`, if that makes it easier.
